### PR TITLE
Adds support for git/repo artifacts

### PR DESCRIPTION
**Pull request by dmrogers7**
Adds support for the `git/repo` artifact type

To test locally

1. Create artifact account config file `/opt/spinnaker/artifacts/config/git-repo.json`
```json
{
  "baseURL": "https://github.com",
  "name": "git-repo",
  "types": [
    "git/repo"
  ]
}
```
2. Start `go-clouddriver`
```
make run
```
3. Make request
```
curl "http://localhost:7002/artifacts/fetch" \
     -s -L -X PUT -H 'Content-Type: application/json' -d '{
	   "version": "master",
	   "artifactAccount": "git-repo",
	   "reference": "https://github.com/billiford/go-clouddriver",
	   "type": "git/repo"
	 }' | tar t -
```

Should see results like
```
.github/
.github/workflows/
.github/workflows/go.yml
.gitignore
Makefile
README.md
api/
api/swagger.yaml
cmd/
cmd/clouddriver/
cmd/clouddriver/clouddriver.go
docker/
docker/Dockerfile
docker/build.sh
...
pkg/sql/sqlfakes/
pkg/sql/sqlfakes/fake_client.go
pkg/task.go
```

### CODE REVIEW

File: pkg/http/core/artifacts.go [hunk 1]
```diff
@@ -212,10 +218,143 @@ func GetArtifact(c *gin.Context) {
...
+			clouddriver.WriteError(c, http.StatusBadRequest, err)
+			return
+		}
+
+		branch := "master"
+		if a.Version != "" {
+			branch = a.Version
+		}
+
+		url := fmt.Sprintf("%s/archive/%s.tar.gz", a.Reference, branch)
```
Comment (billiford) [inline pkg/http/core/artifacts.go:233]: Is there a reason we went with `https://github.com/User/repo/archive/master.tar.gz` over the v3 API `https://api.github.com/repos/User/repo/tarball/master`?

https://stackoverflow.com/questions/8377081/github-api-download-zip-or-tarball-link

**Review by billiford — COMMENTED** [review]
This looks great! See comment.

File: pkg/http/core/artifacts.go (hunk 1 above)
Comment (dmrogers7) [reply pkg/http/core/artifacts.go:233]: No, just less URL manipulation from what the user enters in on the UI
image

Which reminds me - I only added support for token auth, not ssh keys

Comment (billiford) [reply pkg/http/core/artifacts.go:233]: Oh right, that would be a pain to deal with. LGTM - I'll wait on ssh keys (and possibly a test in sandbox, if it's not done already) to merge.

**Pull request closed without merging**